chore(detectors): Add page for Query Injection Issues (#14209)
getsentry/sentry-docs: docs/product/issues/issue-details/query-injection-issues/index.mdx

---
title: "Potential Query Injection Vulnerability Issues"
sidebar_order: 50

> description: "Learn more about Potential Query Injection Vulnerability issues and how to diagnose and fix them."

Reviewer: ```suggestion
description: "Learn more about potential query injection vulnerability issues and how to diagnose and fix them."
```

---

> Potential Query Injection Vulnerability issues are raised when Sentry detects values taken directly from an incoming request being incorporated into a database query. Unsanitised interpolation of user input can lead to [SQL injection](https://owasp.org/www-community/attacks/SQL_Injection) and related attacks.

Reviewer: ```suggestion
Potential Query Injection Vulnerability issues are raised when Sentry detects values taken directly from an incoming request being incorporated into a database query. Unsanitized interpolation of user input can lead to [SQL injection](https://owasp.org/www-community/attacks/SQL_Injection) and related attacks.
```

> ## Detection criteria

Reviewer: ```suggestion
## Detection Criteria
```

The detector evaluates each request in **two stages**:

1. **Filter request values** – Discards tokens that are:

   - too short,
   - SQL keywords, or
   - other frequently benign values

2. **Match against queries** – Scans database queries and if **both** a payload key _and_ its value appear in the same query, Sentry creates a Potential Query Injection Vulnerability issue.

### Example

```
Request →  GET /api?username=bob
Query   →  SELECT * FROM users WHERE username = 'bob'
```

Because the value `'bob'` is inserted directly from the `username` parameter into the query, Sentry flags the operation as potentially vulnerable. **An issue indicates a _risk_, not that an exploit has already occurred.**

### False positives

> Some ORMs or query‑builder libraries assemble SQL strings internally before parameterising them. We suppress many known libraries, but unrecognised ones may still trigger the detector. If you believe an issue is a false positive, leave feedback on the issue page.

Reviewer: ```suggestion
Some ORMs or query‑builder libraries assemble SQL strings internally before parameterizing them. We automatically suppress many known libraries, but unrecognized ones may still trigger the detector. If you believe an issue is a false positive, leave feedback on the issue page.
```
